system/API/ResponseTrait.php

```diff
@@ -328,44 +328,31 @@ protected function format($data = null)
 		// If the data is a string, there's not much we can do to it...
 		if (is_string($data))
 		{
-			$this->setContentType('text/html');
+			// The content type should be text/... and not application/...
+			$contentType = $this->response->getHeaderLine('Content-Type');
+			$contentType = str_replace('application/json', 'text/html', $contentType);
+			$contentType = str_replace('application/', 'text/', $contentType);
```

Comment:
Mainly, I just wanted to bring it up so we could discuss to make sure we weren't missing something. 

I think I didn't notice that it was only for string responses at the time. I just saw the blanket change and got a little nervous. We could got with this for now, I think.